core/src/main/java/com/google/zxing/oned/OneDimensionalCodeWriter.java

```diff
@@ -90,6 +92,14 @@ private static BitMatrix renderResult(boolean[] code, int width, int height, int
     return output;
   }
 
+  /**
+   * Throw IllegalArgumentException if input contains characters other than digits 0-9.
+   */
+  protected static final void checkNumeric(String contents) throws IllegalArgumentException{
+    if (!NUMERIC.matcher(contents).matches()) {
+        throw new IllegalArgumentException("Input should only contain digits 0-9");
```

Comment:
The indentation is off here and in a test below but maybe I can just adjust it later. Change looks OK.

Reply:
Resolved with latest commit